modules/affinity/src/main/java/org/apache/ignite/internal/affinity/Assignments.java

```diff
@@ -0,0 +1,228 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.affinity;
+
+import static org.apache.ignite.internal.affinity.Assignment.forLearner;
+import static org.apache.ignite.internal.affinity.Assignment.forPeer;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+/**
+ * Represents affinity assignments.
+ */
+public class Assignments {
+    private final List<Set<Assignment>> assignments;
+
+    public Assignments(List<Set<Assignment>> assignments) {
+        this.assignments = assignments;
+    }
+
+    public List<Set<Assignment>> assignments() {
+        return assignments;
+    }
+
+    /**
+     * Partitions count.
+     *
+     * @return Partitions count.
+     */
+    public int size() {
+        return assignments.size();
+    }
+
+    /**
+     * Gets an assignments for the given partition.
+     *
+     * @param part Partition.
+     * @return Assignments for the given partition.
+     */
+    public Set<Assignment> get(int part) {
+        return assignments.get(part);
+    }
+
+    /**
+     * Sets an assignment for the given partition.
+     *
+     * @param part Partition.
+     * @param assignment Assignments for the given partition.
+     */
+    public void set(int part, Set<Assignment> assignment) {
+        assignments.set(part, assignment);
+    }
+
+    /**
+     * Bytes representation of the assignments.
+     *
+     * @return Bytes representation of the assignments.
+     */
+    public byte[] bytes() {
+        Map<String, Integer> consistentIds = new LinkedHashMap<>();
+
+        int idx = 0;
+        for (Set<Assignment> assignmentSet : assignments) {
+            for (Assignment a : assignmentSet) {
+                if (consistentIds.putIfAbsent(a.consistentId(), idx) == null) {
+                    idx++;
+                }
+            }
+        }
+
+        byte[] consistentIdsBytes = collectionToBytes(consistentIds.keySet(), String::getBytes);
+
+        byte[] assignmentsBytes = collectionToBytes(
+                assignments,
+                set -> collectionToBytes(set, a -> assignmentToBytes(a, consistentIds::get))
+        );
+
+        ByteBuffer buf = ByteBuffer.allocate(Integer.BYTES * 2 + consistentIdsBytes.length + assignmentsBytes.length);
+
+        buf.putInt(consistentIdsBytes.length);
+        buf.put(consistentIdsBytes);
+        buf.putInt(assignmentsBytes.length);
```

Comment:
I feel like this (`buf.putInt(assignmentsBytes.length)`) is unnecessary, you can calculate the remaining size from what you have already read